samples/snippets/csharp/VS_Snippets_CLR_System/system.idisposable/cs/base1.cs

```diff
@@ -2,25 +2,30 @@
 using System;
 using System.Runtime.InteropServices;
 
-class BaseClassWithSafeHandle : IDisposable
+public class BaseClassWithSafeHandle : IDisposable
 {
     // To detect redundant calls
     private bool _disposedValue;
 
     // Instantiate a SafeHandle instance.
-    private SafeHandle _safeHandle = new SafeFileHandle(IntPtr.Zero, true);
+    private SafeHandle? _safeHandle = new SafeFileHandle(IntPtr.Zero, true);
 
     // Public implementation of Dispose pattern callable by consumers.
-    public void Dispose() => Dispose(true);
+    public void Dispose()
+    {
+        Dispose(true);
+        GC.SuppressFinalize(this);
+    }
 
     // Protected implementation of Dispose pattern.
     protected virtual void Dispose(bool disposing)
     {
         if (!_disposedValue)
         {
-            if (disposing)
+            if (disposing && _safeHandle is not null)
             {
                 _safeHandle.Dispose();
```

Comment:
I generally prefer this, but I don't have strong feelings:

```suggestion
            if (disposing)
            {
                _safeHandle?.Dispose();
```